### ISSUE TITLE
Support for dynamic URL params as variables in Formhandler vega charts spec

### ISSUE THREAD

**Issue by tejesh95** [gramener/gramex#99]
```yaml
url:
  formhandler-vega-1:
    pattern: /$YAMLURL/vega-1
    handler: FormHandler
    kwargs:
      url: $YAMLPATH/flags.csv?METRIC=C1
      formats:
        barchart:           # Allows ?_format=barchart
          format: vega
          spec:
            "$schema": "https://vega.github.io/schema/vega/v4.json"
            .......
            .......
            - name: xscale
                type: band
               domain:
                   data: table
                   field: {METRIC}

```

raised by @kriti21

**Comment by tejesh95** [comment]
work around, is to use formhandler modify method to create a `dummy_column` which stores the METRIC  values in it.

**Comment by joshuamosesb** [comment]
```
...
    handler: FormHandler
    kwargs:
      url: $data_file
      default:
        _format: custom
        _chtype: point
        _xcol: sindpop
        _ycol: fsize
        _cate: cid
        _xcoltype: quantitative
        _ycoltype: quantitative
      formats:
        coloredscatterplot:
          format: vega-lite
          spec:
            "$schema": "https://vega.github.io/schema/vega-lite/v2.json"
            width: 380
            height: 400
            autosize:
              type: fit
              contains: padding
            mark: point
            encoding:
              y: {field: fsize, type: quantitative}
              x: {field: sindpop, type: quantitative}
              color: {field: cid, type: nominal}
        custom:
          format: vega-lite
          spec:
            "$schema": "https://vega.github.io/schema/vega-lite/v2.json"
            autosize:
              type: fit
              contains: padding
            width: 400
            height: 380
            mark: '{chtype}'
            encoding:
              x: {field: '{xcol}', type: '{xcoltype}'}
              y: {field: '{ycol}', type: '{ycoltype}'}
              color: {field: '{cate}', type: nominal}
```
is it possible to use URL-params _xcol, _ycol to dynamically generate vega-lite charts defined with reference to '{xcol}' & '{ycol}' etc,. ?

In the above snippet `_format=coloredscatterplot` is rendering fine, but `_format=custom` is showing an error: `TypeError: Cannot read property 'orient' of undefined`.
Unable to trace the error in PY/JS console.

**Comment by pratapvardhan** [comment]
Closed via 9a0b579
